### PR TITLE
Update travis-ci script so that PRs can fail due to pester tests

### PR DESCRIPTION
**Pull request by TravisEz13**
FYI @markekraus 

Test-PSPesterResults was expected to throw an error by the travis.ps1.  It was only writing to the host.  

I've updated Test-PSPesterResults to throw an error after it writes to the host.

I've also added a switch to Start-PSPester which will inject a failing test and the [includeFailingTest] in travis.ps1 which will use this switch so we can verify that it works.

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@TravisEz13 can you please briefly describe the issue, root cause and explain the fix? That would be helpful to the review.

**Comment by TravisEz13** [comment]
@daxian-dbw updated the description.

File: tools/failingTests/fail.tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Describe "Failing test used to test CI Scripts" {
+    It "Should fail" {
+        1 | should be 2
+    }
+}
```
Comment (TravisEz13) [inline tools/failingTests/fail.tests.ps1]: I'll add the line ending later.  I'll use this change to remove the tag that forces a test failure.

File: tools/travis.ps1 [hunk 2]
```diff
@@ -203,16 +220,6 @@ else
...
-        # Determine whether the build passed
-        try {
-            # this throws if there was an error
-            Test-PSPesterResults -ResultObject $pesterPassThruObject
-            $result = "PASS"
-        }
-        catch {
-            $resultError = $_
-            $result = "FAIL"
-        }
```
Comment (daxian-dbw) [inline tools/travis.ps1]: Isn't this change already done in #4807?

Comment (TravisEz13) [reply tools/travis.ps1]: I rebased and resolved the issue.

**Comment by TravisEz13** [comment]
Verified tests fail correctly here:  https://travis-ci.org/PowerShell/PowerShell/jobs/275168248

**Review by daxian-dbw — APPROVED** [review]

File: build.psm1 [hunk 3]
```diff
@@ -843,9 +843,15 @@ function Start-PSPester {
         [switch]$Unelevate,
         [switch]$Quiet,
         [Parameter(ParameterSetName='PassThru',Mandatory=$true)]
-        [switch]$PassThru
+        [switch]$PassThru,
+        [switch]$IncludeFailingTest
     )
 
+    if($IncludeFailingTest.IsPresent)
```
Comment (markekraus) [inline build.psm1]: space between `if` and `(`

File: tools/travis.ps1 [hunk 4]
```diff
@@ -167,6 +168,11 @@ else
         $pesterParam['ThrowOnFailure'] = $true
     }
 
+    if($hasRunFailingTestTag)
+    {
```
Comment (markekraus) [inline tools/travis.ps1]: space between `if` and `(`

**Review by markekraus — COMMENTED** [review]
2 minor formatting comments, otherwise LGTM

File: tools/travis.ps1 (hunk 4 above)
Comment (TravisEz13) [reply tools/travis.ps1]: Resolved

File: build.psm1 (hunk 3 above)
Comment (TravisEz13) [reply build.psm1]: Resolved

**Review by markekraus — APPROVED** [review]

**Pull request merged**